src/shapes/object.class.js

```diff
@@ -1729,16 +1729,16 @@
         canvas.backgroundColor = '#fff';
       }
       this.setPositionByOrigin(new fabric.Point(canvas.width / 2, canvas.height / 2), 'center', 'center');
-
       var originalCanvas = this.canvas;
-      canvas.add(this);
+      canvas._objects = [this];
+      this.set('canvas', canvas);
       var canvasEl = canvas.toCanvasElement(multiplier || 1, options);
-      this.shadow = originalShadow;
       this.set('canvas', originalCanvas);
+      this.shadow = originalShadow;
       if (originalGroup) {
         this.group = originalGroup;
       }
-      this.set(origParams).setCoords();
```

Comment:
please let's leave both setCoords.
Now it should not matter, but we know correct coords are necessary for renders.
if something changes later or in an after:render loop, since this object to canvas export is a bit of an hack in which the object gets added to a canvas to be rendered, let's be sure we set coords correctly when we finish the procedure.
Let's not optimize when we know we are fragile in those kinds of operations. The example is the exact regression we are fixing, we introduced because we did not remember of this.
Let's not overdo and let's leave setCoords in place.

Reply:
Good point

Reply:
a2294f9b